lua5_3/lua5_3.l

```diff
@@ -0,0 +1,63 @@
+%%
+and "AND"
+break "BREAK"
+do "DO"
+else "ELSE"
+elseif "ELSEIF"
+end "END"
+false "FALSE"
+for "FOR"
+function "FUNCTION"
+goto "GOTO"
+if "IF"
+in "IN"
+local "LOCAL"
+nil "NIL"
+not "NOT"
+or "OR"
+repeat "REPEAT"
+return "RETURN"
+then "THEN"
+true "TRUE"
+until "UNTIL"
+while "WHILE"
+[a-zA-Z_][a-zA-Z_0-9]* "NAME"
+\+ "+"
+\- "-"
+\* "*"
+/ "/"
+% "%"
+\^ "^"
+# "#"
+& "&"
+~ "~"
+\| "|"
+<< "<<"
+>> ">>"
+// "//"
+== "=="
+~= "~="
+<= "<="
+>= ">="
+< "<"
+> ">"
+= "="
+\( "("
+\) ")"
+\{ "{"
+\} "}"
+\[ "["
+\] "]"
+:: "::"
+; ";"
+: ":"
+, ","
+("(\\[abfnrtvz\\"']|\\x[0-9a-fA-F]|\\[0-9][0-9]?[0-9]?|\\u\{[0-9a-fA-F]+\}|.)*?")|('(\\[abfnrtvz\\"']|\\x[0-9a-fA-F]|\\[0-9][0-9]?[0-9]?|\\u\{[0-9a-fA-F]+\}|.)*?') "SHORT_STR"
+(\[\[.*?\]\])|(\[=\[.*?\]=\])|(\[==\[.*?\]==\]) "LONG_STR"
+(0[xX][0-9a-fA-F]+(\.[0-9a-fA-F]+)?([pP]-?[0-9a-fA-F]+)?)|(\.[0-9]+([eE]-?[0-9]+)?)|([0-9]+(\.[0-9]*)?([eE]-?[0-9]+)?) "NUMERAL"
+(--\[=\[.*?\]=\])|(--\[\[.*?\]\])|(--.*?$) ;
```

Comment:
If I'm reading this right this is the line that is trying to parse comments. What i was talking about was the number of equals between the between the opening brackets  can be more than one
``` Lua
--[==[]=]true]==]a = b
```
and it has to be ended with closing brackets that have the same number of equals.

Reply:
Correct on all fronts. Unfortunately, with normal (non-Perl) regular expressions we can't express "match '='s at the front and end". However, what I can do (which I've done in the most recent commit) is allow up to $n$ '='s. In this case I've allowed 5, which I suspect is more than anyone ever uses in practise. Do you think that's enough (or maybe even too many!)?

Reply:
i don't think you would normally need to go over two most the time.

Reply:
OK, then I think 5 is plenty :)